pkg/converter/pod_converter.go

```diff
@@ -26,13 +26,39 @@ import (
 // Label which represents the namespace a given pod belongs to.
 const k8sNamespaceLabel = "calico/k8s_ns"
 
+// WorkloadEndpointData is an internal struct used to store the various bits
+// of information that the policy controller cares about on a workload endpoint.
+type WorkloadEndpointData struct {
+	Key    string
+	Labels map[string]string
+}
+
 type podConverter struct {
 }
 
+// BuildWorkloadEndpointData generates the correct WorkloadEndpointData for the given
+// WorkloadEndpoint, extracting fields that the policy controller is responsible for syncing.
+func BuildWorkloadEndpointData(wep api.WorkloadEndpoint) WorkloadEndpointData {
+	return WorkloadEndpointData{
+		Key:    wep.Metadata.Workload,
+		Labels: wep.Metadata.Labels,
+	}
+}
+
+// MergeWorkloadEndpointData applies the given WorkloadEndpointData to the provided
+// WorkloadEndpoint, updating relevant fields with new values.
+func MergeWorkloadEndpointData(wep *api.WorkloadEndpoint, upd WorkloadEndpointData) {
+	if wep.Metadata.Workload != upd.Key {
+		log.Fatalf("Bad attempt to merge data for %s into wep %s", upd.Key, wep.Metadata.Workload)
+	}
+	wep.Metadata.Labels = upd.Labels
```

Comment:
this is where namedPorts update will happen?

Reply:
Yep